source/unified-test-format/unified-test-format.rst

```diff
@@ -462,6 +477,78 @@ The structure of this object is as follows:
 
     Test files SHOULD NOT use this option unless one or more command monitoring
     events are specified in `observeEvents <entity_client_observeEvents_>`_.
+    
+  - ``storeEventsAsEntities``: Optional map of event names to entity names.
+    If provided the test runner MUST:
+    
+      - For each entity name, create the respective entity with a type of
+        "event list". If multiple event names map to the same entity name,
+        exactly one entity MUST be created. If the entity already exists
+        (such as from a previous ``storeEventsAsEntities`` declaration from
+        another client), the test runner MUST raise an error.
+      - Set up an event subscriber for each event named. The event subscriber
+        MUST serialize the events it receives into a document and append
+        the document to the array stored in the specified entity.
+        The fields in the document MUST be as follows:
+        
+        - For ``CommandStartedEvent``:
+       
+          - ``name``: the name of the event, i.e. ``CommandStartedEvent``.
+          - ``commandName``: the name of the command, e.g. ``insert``.
+          - ``startTime``: the (floating-point) number of seconds since
+            the Unix epoch when the command began executing.
+          - ``address``: the address of the server to which the command
+             was sent, e.g. ``localhost:27017``.
+        
+        - For ``CommandSucceededEvent``:
+       
+          - ``name``: the name of the event, i.e. ``CommandSucceededEvent``.
+          - ``commandName``: the name of the command, e.g. ``insert``.
+          - ``duration``: the time, in (floating-point) seconds, it took for
+            the command to execute.
+          - ``startTime``: the (floating-point) number of seconds since
+            the Unix epoch when the command began executing.
+          - ``address``: the address of the server to which the command
+             was sent, e.g. ``localhost:27017``.
+        
+        - For ``CommandFailedEvent``:
+       
+          - ``name``: the name of the event, i.e. ``CommandFailedEvent``.
+          - ``commandName``: the name of the command, e.g. ``insert``.
+          - ``duration``: the time, in (floating-point) seconds, it took for
+            the command to execute.
+          - ``failure``: this field MUST contain a textual description
+             of the error encountered while executing the command.
+          - ``startTime``: the (floating-point) number of seconds since
+            the Unix epoch when the command began executing.
+          - ``address``: the address of the server to which the command
+             was sent, e.g. ``localhost:27017``.
+             
+        - For connection pool events (``PoolCreatedEvent``,
```

Comment:
Adding this will also require that we discuss how to enable CMAP and revise the current advice not to mix different types of observed events. IIRC, I also suggested coordinating with someone from the CMAP spec (e.g. @patrickfreed) so the approach taken here will also work with that project migrating to the unified format (assuming that's still planned).

Reply:
To clarify, this PR does not need CMAP integrated into it in the same way we currently handle command monitoring (i.e. `observeEvents` and 'expectEvents`). @p-mongo introduced syntax to client entity creation to collect CMAP events and append them to a list/array in the entity map where they can be retrieved by some other code (e.g. workload executor, which invokes the test runner).

It's understandable if there's no need/benefit to port CMAP tests to the unified test runner. I haven't worked with them and didn't now they were possible unit tests.

Looking at the [SDAM integration tests README](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/tests/README.rst), I see some reference to CMAP events but it looks like they're mostly queried through special operations (not like `expectEvents`). If/when that spec is ported to the unified test runner, we can revisit this.

I'll resolve this thread. Thanks for the feedback.